### PR TITLE
1174 set log level to debug in golang

### PR DESCRIPTION
**Pull request by alonsopec89**
Tested manually and seems like works

Checklist (if applicable):
- [X] Tested (manually, unit tested, etc.)
- [ ] Docs updated

### CODE REVIEW

**Comment by apascal07** [comment]
We want to go through all of the log statements we have today and change non-essential stuff to DEBUG instead of INFO like it is today. We still want to log all of the levels if the right argument is passed.

File: go/core/logger/logger.go
Comment (apascal07) [inline go/core/logger/logger.go:1]: Do we need the changes in this file?

Comment (alonsopec89) [reply go/core/logger/logger.go:1]: I just moved to a new file

File: go/core/logger/logger.go [hunk 1]
```diff
@@ -26,10 +26,12 @@ import (
 func init() {
 	// TODO: Remove this. The main program should be responsible for configuring logging.
```
Comment (apascal07) [inline go/core/logger/logger.go:27]: Has this TODO been resolved? How does the log level get configured by the user?

Comment (alonsopec89) [reply go/core/logger/logger.go]: yes, just has been resolved I think

Comment (alonsopec89) [reply go/core/logger/logger.go]: yes, It is resolved, the user can set the log level like:

```go
if err := genkit.Init(ctx, &genkit.Options{LogLevel: slog.LevelWarn}); err != nil {
        log.Fatal(err)
}
```

Comment (apascal07) [reply go/core/logger/logger.go]: Hmm, this is a bit tangential to the logging. In theory, flows/etc can be used without calling `genkit.Init` which is mostly responsible for starting the servers, which can be sidestepped by starting up your own server and exposing the flows directly. We still want to configure the logging level appropriately in those cases. In JS, we pulled it out of Genkit instance config and it's a top-level/global function.

Comment (alonsopec89) [reply go/core/logger/logger.go]: I refactored it, now we can set the log level like:

```go
logger.SetLevel(slog.LevelError)
```

and the default log level now is: LevelDebug

**Review by apascal07 — APPROVED** [review]

**Pull request merged**